cloudinit/config/cc_ca_certs.py

```diff
@@ -67,13 +67,19 @@ def add_ca_certs(certs):
         cert_file_contents = "\n".join([str(c) for c in certs])
         util.write_file(CA_CERT_FULL_PATH, cert_file_contents, mode=0o644)
 
-        # Append cert filename to CA_CERT_CONFIG file.
-        # We have to strip the content because blank lines in the file
-        # causes subsequent entries to be ignored. (LP: #1077020)
-        orig = util.load_file(CA_CERT_CONFIG)
-        cur_cont = '\n'.join([line for line in orig.splitlines()
-                              if line != CA_CERT_FILENAME])
-        out = "%s\n%s\n" % (cur_cont.rstrip(), CA_CERT_FILENAME)
```

Comment:
Would a simpler fix be to add an `out = out.strip()` to ensure that any leading/trailing newlines are removed in all cases?

Reply:
Are you suggesting adding a new line 77 as "out = out.strip()"? That wouldn't work - as well as removing any newline at the start of the file it would also remove the final newline at the very end of the file (after CA_CERT_FILENAME) which I believe is needed and certainly the testcases test_single_cert_trailing_cr and test_single_cert_no_trailing_cr check for it.

Adding a line 77 with a lstrip() instead would do the job would it feels find of hacky adding a leading newline (line 76) and then removing it (line 77) for the empty file case.

The reason I added the if/else around lines 73-75 as well as line 76, rather than just around line 76, was that it felt inefficient having line 73 opening and loading a file and lines 74,75 trying to process its contents when the file is empty.

Anyway I'm still trying to get a handle on the cloud-init project's code standards so let me know the recommended way to proceed. I raised this PR and have another couple more to raise as "training" to go through a few reviews here before I submit my substantial PR for Alpine Linux support.

Reply:
@OddBloke any comments?

Reply:
Your reasoning sounds good to me, thanks!